atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/PersistentLockManager.java

```diff
@@ -101,9 +101,11 @@ public synchronized void releasePersistentLock() {
             persistentLockService.releaseBackupLock(lockId);
             lockId = null;
         } catch (CheckAndSetException e) {
-            log.error("Failed to release persistent lock {}. "
-                    + "Either the lock was already released, or communications with the database failed.",
+            log.error("Failed to release persistent lock {}. The lock must have been released from under us. "
+                            + "If you didn't release it manually, this warrants investigation, but future sweeps "
+                            + "should correctly be able to re-acquire the lock.",
                     SafeArg.of("lock id", lockId), e);
+            lockId = null;
```

Comment:
At this point it's probably okay to move `lockId = null` out of the try-catch block, since it is always cleared on a release(). I'm fine either way though